README.md

```diff
@@ -138,6 +138,19 @@ Image reprojection and warping utility. This is the equivalent of the [gdalwarp]
 #### Return value
 A promise that resolves to a new `GDALDataset`.
 
+<br />
+
+### `GDALDataset.render(mode, args, colors)`
+Utility for rendering and computing DEM metrics. This is the equivalent of the [gdaldem](https://gdal.org/programs/gdaldem.html) command.
+
+**Note**: This returns a new `GDALDataset` object but does not perform any immediate calculation. Instead, calls to `.render()` are evaluated lazily (as with `convert()` and `warp()`, above). The render operation is only evaluated when necessary in order to access some property of the dataset, such as its size, bytes, or band count. Successive calls to `.warp()` and `.convert()` can be lazily chained onto datasets produced by `.render()`, and vice-versa.
+#### Parameters
+- `mode`: One of ['hillshade', 'slope','aspect', 'color-relief', 'TRI', 'TPI', 'roughness']. See the [`gdaldem documentation`](https://gdal.org/programs/gdaldem.html#cmdoption-arg-mode) for an explanation of the function of each mode.
+- `args`: An array of strings, each representing a single [command-line argument](https://gdal.org/programs/gdaldem.html#synopsis) accepted by the `gdaldem` command. The `inputdem` and `output_xxx_map` parameters should be omitted; these are handled by `GDALDataset`. Example: `ds.render('hillshade', ['-of', 'PNG'])`
```

Comment:
Ensuring that `inputdem` and `output` aren't in this array when a user calls the function doesn't seem to really be a thing that could be validated in code, right? You'd just be looking for a string filename and presumably if you passed them in `args` the existing error handling would throw with a somewhat helpful error?

Reply:
This is a good question -- I don't know what will happen if the user erroneously puts in an input or output filename. I'll write a test for it and see what happens. But yeah, my hope is that the GDAL argument parsing will be able to provide helpful feedback in this case, because it would basically end up with two of them.

Reply:
Looks like it says something along the lines of `Too many command options 'input.tif'`, which is not totally clear about what's going on but at least identifies the problematic argument correctly. I think seeing that would drive most users to try removing the `input.tif` option as a first step, and that would be the right thing to do even though the explanation wouldn't have reinforced that the filenames are handled internally.

Reply:
As a note, I'm not including the test I wrote because all the relevant functionality is on the GDAL side so the test wouldn't exercise any Loam code that doesn't already have a test.

Reply:
That error seems to be good enough for now. If this trips people up in the future, we could look for the "Too many command options" part of the error message and append something like "Be sure you're not including your input and output tif paths to args." to it. Otherwise, like you said, doesn't seem to be much can be done.